src/main/java/de/tum/in/www1/artemis/domain/Submission.java

```diff
@@ -110,12 +112,54 @@ public Result getLatestResult() {
         return null;
     }
 
+    /**
+     * Used to get result by correction round
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRound(Long correctionRound) {
+        if (results != null && results.size() > correctionRound) {
+            return results.get(correctionRound.intValue());
+        }
+        return null;
+    }
+
+    /**
+     * Used to get result by correction round ignoring all automatic results
+     * @param correctionRound to get result by
+     * @return the result based on the given correction-round
+     */
+    @Nullable
+    @JsonIgnore
+    public Result getResultByCorrectionRoundIgnoreAutomatic(Long correctionRound) {
+        List<Result> withoutAutomaticResults = results.stream().filter(result -> result != null && !result.getAssessmentType().equals(AssessmentType.AUTOMATIC))
+                .collect(Collectors.toList());
+        if (withoutAutomaticResults != null && withoutAutomaticResults.size() > correctionRound) {
+            return withoutAutomaticResults.get(correctionRound.intValue());
+        }
+        return null;
+    }
+
     @Nullable
     @JsonProperty(value = "results", access = JsonProperty.Access.READ_ONLY)
     public List<Result> getResults() {
         return results;
     }
 
+    @Nullable
+    @JsonIgnore
+    public List<Result> getManualResults() {
+        return results.stream().filter(result -> result != null && !result.getAssessmentType().equals(AssessmentType.AUTOMATIC)).collect(Collectors.toList());
+    }
+
+    @Nullable
+    @JsonIgnore
+    public List<Result> getManualAndNullResults() {
```

Comment:
this does not really make sense to me. Why should you add null values to the list?
And why would you want to retrieve those?

Reply:
The null values left in the list are the results of other correctionRounds assessed by different tutors. So after this we get a list, where only at the positions where the current tutor has assessed is a result that is not null. 
Instead of doing this in the Submission.java we could also just use the stream directly in the `getAllProgrammingSubmissionsAssessedByTutorForCorrectionRoundAndExercise` where it's called

Reply:
I still don't understand it. There should never be `null` values in a list. Why should other correction rounds lead to such values? Then something is wrong in the other place

Reply:
Ok I just found https://stackoverflow.com/questions/13307849/hibernate-returns-list-with-null-values-onetomany-annotation-with-list-type
This seems to be the normal behavior when filtering values in the query. To maintain the "correct" order, Hibernate adds null values. I think we should document this here in JavaDoc because this is not very intuitive, but makes sense.

I would still prefer a different handling here. Instead of returning all those lists, please think about the actual purpose. If you want to get a specific result, why don't you add methods such as 

`getResultForCorrectionRound(int correctionRound)`